libafl/src/executors/inprocess/stateful.rs

```diff
@@ -86,16 +86,23 @@ where
         input: &I,
     ) -> Result<ExitKind, Error> {
         *state.executions_mut() += 1;
+
+        // unsafe operations for settings up pointers
```

Comment:
Can't `run_target` still be called from multiple threads in an unsafe way, here?

We should probably still check that the globals are not set

Reply:
then better make run_target unsafe

Reply:
It'll trickle up though, right? 
In general this can be made safe I think. We coudl do what Addison wanted to do and give out a token. That way it cannot be used twice at the same time(?)

Reply:
we should make it thread local value

Reply:
>  We coudl do what Addison wanted to do and give out a token. That way it cannot be used twice at the same time(?)

i don't think so.
as long as you have two threads data races will occur.
there are just two solutions. 
you put a semaphore here or make it thread local